core/domain/platform_parameter_domain.py

```diff
@@ -194,35 +232,46 @@ def from_dict(cls, client_context_dict, server_context_dict):
             object.
         """
         return cls(
-            client_context_dict.get('platform_type'),
+            client_context_dict['platform_type'],
             client_context_dict.get('browser_type'),
             client_context_dict.get('app_version'),
-            server_context_dict.get('server_mode'),
+            server_context_dict['server_mode'],
```

Comment:
Only `platform_type` and `server_mode` is converted because:
1. in codebase there are some cases where we are not providing `browser_type` key or `app_version` key. for eg: https://github.com/oppia/oppia/blob/3f1af7c7c7c418d836b03824e9bb72ca705d11ce/core/domain/platform_feature_services.py#L165
2. Also, after looking at the frontend i found we are not providing the `app_version` key ( Please correct me if I'm wrong here). Link: https://github.com/oppia/oppia/blob/3f1af7c7c7c418d836b03824e9bb72ca705d11ce/core/templates/domain/platform_feature/client-context.model.ts#L48
3. In many test cases either we are not providing `browser_type` key or `app_version` key.

Reply:
Thanks for the note. This may be worth adding as a comment/TODO here because, if you look at the code you linked in (1), it seems that we do eventually want to introduce an app_version (at least). So maybe worth writing something similar here about changing the .get() to [] once the corresponding TODO is resolved (so that a dev who searches for the relevant TODO number in the future can find this too).

Reply:
Done TODO is added!!